### PR TITLE
feat: "Start Agent" button fix 

### PR DESCRIPTION
**Pull request by mohandast52**
- "Start Agent" button was repeated three times, causing the loading and other states to be deprioritized due to overlapping conditions.
- Refactored the code to have a single "Start Agent" button at the bottom and added props in the condition.
- It should work exactly the same as before and especially show different states when the "Start Agent" button is clicked **_after eviction_**.

### CODE REVIEW

File: frontend/components/Main/MainHeader/index.tsx [hunk 1]
```diff
@@ -274,27 +274,20 @@ export const MainHeader = () => {
...
     const serviceExists = !!services?.[0];
 
-    if (!isDeployable) {
-      return (
-        <Button type="default" size="large" disabled>
-          Start agent {!serviceExists && '& stake'}
-        </Button>
-      );
-    }
+    const canActivateAgent = canStartAgent && (isAgentEvicted || isDeployable);
```
Comment (truemiller) [inline frontend/components/Main/MainHeader/index.tsx:279]: getting lost with these `can` variables, they need more context

i.e. `canStartAgent` is not solely determined by the staking contract, and needs to be renamed to something more fitting, unsure what exactly but it should infer it's relation to the staking contract

`canActivateAgent`, i'm lost on what this represents too :/ canStartAgent already factors in isAgentEvicted

image

**Review by truemiller — COMMENTED** [review]
looks good but need to rename appropriately so it's more obvious what state we're setting conditions against 🙏

File: frontend/components/Main/MainHeader/index.tsx [hunk 2]
```diff
@@ -274,27 +277,17 @@ export const MainHeader = () => {
...
-      return (
-        <Button type="default" size="large" disabled>
-          Start agent {!serviceExists && '& stake'}
-        </Button>
-      );
-    }
+    const buttonProps: ButtonProps = {
+      type: 'primary',
+      size: 'large',
+      disabled: !isEligibleForStakingAction || !isDeployable,
```
Comment (Tanya-atatakai) [inline frontend/components/Main/MainHeader/index.tsx:285]: don't we check the `isEligibleForStakingAction` condition already in `if (!isEligibleForStakingAction) return <CannotStartAgent />;`?

File: frontend/components/Main/MainHeader/index.tsx [hunk 3]
```diff
@@ -274,27 +277,17 @@ export const MainHeader = () => {
...
+      onClick: canStartEvictedAgent || isDeployable ? handleStart : undefined,
```
Comment (Tanya-atatakai) [inline frontend/components/Main/MainHeader/index.tsx:286]: I wonder, would it improve readability if we move `canStartEvictedAgent` into `isDeployable`?

**Review by truemiller — APPROVED** [review]

**Review by Tanya-atatakai — APPROVED** [review]

**Review by oaksprout — APPROVED** [review]

**Pull request merged**